pkg/sync/sync_context.go

```diff
@@ -522,6 +559,24 @@ func (sc *syncContext) getSyncTasks() (_ syncTasks, successful bool) {
 		}
 	}
 
+	//namespace auto-creation is created by pre-sync task. This task's liveObj is always nil since it did not go through reconciliation.
```

Comment:
The user might have a custom health check for namespace resource, so the namespace creation task is not necessarily successful when the namespace is created.  I think it is better to just load the actual live resource and at line [536](https://github.com/argoproj/gitops-engine/pull/94/files#diff-f6928024340969ce58a20e045c7dc520R536) an save it in `task.liveObj` field.